### PR TITLE
Average age function was implemented

### PR DESCRIPTION
**Pull request by yuliia-dronova**

### CODE REVIEW

**Review by blcts — APPROVED** [review]
wow I like it 😍

File: src/averageAges.js [hunk 1]
```diff
@@ -20,6 +20,20 @@ function calculateMenAverageAge(people, century) {
...
+
+  const menOnly = filterBySex(people, 'm');
+
+  const menOnlyCentury = menOnly.filter((person) => (
+    Math.ceil(person.died / 100) === century)
+  );
+
+  let menOnlyAverageAge;
+
+  isFinite(century)
```
Comment (ivan-kadykalo) [inline src/averageAges.js:32]:
```suggestion
  const menOnlyAverageAge = isFinite(century)
```

File: src/averageAges.js [hunk 2]
```diff
@@ -37,7 +51,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenOnly = filterBySex(people, 'f');
+  const womenOnlyHasChild = filterMothers(people);
+
+  let womenOnlyAverageAge;
+
+  withChildren
```
Comment (ivan-kadykalo) [inline src/averageAges.js:59]:
```suggestion
  const womenOnlyAverageAge = withChildren
```

File: src/averageAges.js [hunk 3]
```diff
@@ -55,11 +78,45 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = filterMothers(people);
+  const sonsAndDaughters = filterHasMother(people);
+  const sonsOnly = filterBySex(sonsAndDaughters, 'm');
+  let children;
+
+  onlyWithSon
```
Comment (ivan-kadykalo) [inline src/averageAges.js:86]:
```suggestion
  const children = onlyWithSon
```

**Review by ivan-kadykalo — APPROVED** [review]
💪

**Pull request closed without merging**